src/debugger/appWorker.ts

```diff
@@ -186,8 +189,18 @@ export class MultipleLifetimesAppWorker {
         socketToApp.on("message",
             (message: any) => this.onMessage(message));
         socketToApp.on("error",
-            (error: Error) => printDebuggingError("An error ocurred while using the socket to communicate with the React Native app", error));
-        return socketToApp;
+            (error: Error) => {
+                if (retryAttempt) {
+                    Log.logWarning(ErrorHelper.getNestedWarning(error, "Reconnection to the proxy (Packager) failed. Please check the output window for Packager errors, if any. If failure persists, please restart the React Native debugger."));
+                }
+
+                deferred.reject(error);
+            });
+
+        // In an attempt to catch failures in starting the packager on first attempt,
+        // wait for 300 ms before resolving the promise
+        Q.delay(300).done(() => deferred.resolve(socketToApp));
+        return deferred.promise;
```

Comment:
We should change this method to return { WebSocket, Q.Promise<void>}. This code has a potential race condition if the application starts to run before the 300 ms, because the this.socketToApp.send(stringified); line will fail.